[STORY-1877] Introducing the new Limited Collaborator role (#3212)
Scalingo/documentation: src/_posts/platform/user-management/2000-01-01-roles.md

---
title: Roles
modified_at: 2025-06-25 00:00:00
tags: user role permission collaborator
index: 3
---

Scalingo provides three predefined roles to help you adapt permissions to the level of responsibility you want to delegate to your collaborators:

* **Owner**: Reserved for the application owner, this role grants the highest level of permissions, including full administrative control such as deleting or transferring the application. The Owner is also the one who is billed for the application’s usage.  
* **Collaborator**: Best suited for trusted team members who require full operational and data access, including the ability to manage other collaborators.  

> * **Limited** **Collaborator**: Ideal for developers or technical contributors who need to operate applications without accessing sensitive data or modifying critical settings.

Reviewer: ```suggestion
* **Limited** **Collaborator**: Ideal for developers or technical contributors who need to take part to the application life cycles without accessing sensitive data or modifying critical settings.
```

## Role Capabilities Overview

> The following table helps compare roles. **Limited Collaborator** is the safest default for new users.

Reviewer: ```suggestion
The following table helps comparing roles. **Limited Collaborator** is the safest role for new users.
```

| Role | Admin Control | App Configuration | Deployment | Add-ons Management | Collaborator Management | Access Data & Secrets |
| :---- | :---- | :---- | :---- | :---- | :---- | :---- |
| Owner | ✅ Full | ✅ Full | ✅ Full | ✅ Ful | ✅ Full | ✅ Full |
| Collaborator | 🚫 None | ✅ Full | ✅ Full | ✅ Full | ✅ Full | ✅ Full |
| Limited Collaborator | 🚫 None | 🚫 None | ⚠️ Limited | 🚫 None | 🚫 None | 🚫 None |

> This table provides a simplified overview of the main permissions associated with each role. For a complete list of access rights please refer to the [collaborator permissions matrix]({% post_url platform/user-management/2000-01-01-permission-matrix %}).

Reviewer: ```suggestion
This table provides a summarized overview of the main permissions associated with each role. For a complete list of access rights please refer to the [collaborator permissions matrix]({% post_url platform/user-management/2000-01-01-permission-matrix %}).
```

{% note %}  
For improved security, especially in larger organizations, we recommend reviewing our [security best practices guide]({% post_url platform/user-management/2000-01-01-guides %}).  
{% endnote %}

## Owner

> Reserved for the application owner, this role grants the highest level of permissions, including full administrative control such as deleting or transferring the application. The Owner is also the one who is billed for the application’s usage.

Reviewer: ```suggestion
Reserved for the application owner, this role is the most permissive, including full administrative control such as deleting or transferring the application. The Owner is also the one who is billed for the application’s usage.
```

## Collaborator

> The Collaborator role is intended for trusted team members who are directly responsible for the configuration, deployment, and operational maintenance of applications and databases. It grants broad permissions, including access to environment variables, hosted data, deployment settings, and collaborator management, making it suitable for lead developers or DevOps engineers who need full technical control over the application.

Reviewer: ```suggestion
The Collaborator role is intended for trusted team members who are directly responsible for the configuration, deployment and operational maintenance of applications and databases. It grants broad permissions, including access to environment variables, hosted data, deployment settings, and collaborators management. It's a well suited role for lead developers or DevOps engineers, who need full technical control over the application.
```

While collaborators can perform almost all actions, including inviting and managing other users, they cannot delete, rename, or transfer the application. Those actions remain reserved for the Owner.

{% note %}  
This role should only be assigned to individuals you fully trust, as it allows unrestricted access to sensitive data and the ability to make impactful changes to the application environment.  
{% endnote %}

## Limited Collaborator

> The Limited Collaborators role is designed to minimize the risks associated with granting access to third parties or collaborators without operational responsibility. It allows contributors to perform essential day-to-day operations such as restarting applications, viewing logs or metrics, and managing review apps while preventing actions that could compromise the stability, confidentiality, or integrity of your services.

Reviewer: ```suggestion
The Limited Collaborators role is designed to grant access to third parties or collaborators with a limited set of operational actions. It allows contributors to perform essential day-to-day operations such as restarting applications, viewing logs or metrics, and managing review apps while preventing actions that could compromise the stability, confidentiality, or integrity of your services.
```

This role explicitly restricts access to critical operations like modifying configurations, deploying unvalidated code, exploring or exporting sensitive data and managing collaborators.

It is especially suited for environments where separation of duties is required. It ensures that users can contribute to the project without being able to compromise service stability, data integrity, or team governance.

{% note %}  
In most cases, we recommend assigning the Limited Collaborator role by default, unless the user explicitly needs full access to configuration, data, or collaborator management.  
{% endnote %}

## Threat Modeling and Shared Responsibility

> Roles on Scalingo are designed to enforce the principle of least privilege within the platform. They control what users can see or modify across applications, databases, deployments, and configuration layers. By assigning roles appropriately, you can reduce the risk of accidental changes or unauthorized access to critical resources.

Reviewer: ```suggestion
Roles on Scalingo are designed to enforce the principle of least privilege within the platform. They control what users can see or modify accross applications, databases, deployments, and configuration layers. By assigning roles appropriately, you can reduce the risk of accidental changes or unauthorized access to critical resources.
```

> However, platform-level access control is only one layer of protection. Application integrity also depends on the quality and trustworthiness of the code being deployed. Scalingo does not inspect your source code or development workflows. As such, it is the responsibility of each owner to:

Reviewer: ```suggestion
However, platform-level access control is only one layer of protection. Application integrity is also depending on the quality and trustworthiness of the code being deployed. Scalingo does not inspect your source code or development workflows. As such, it is the responsibility of each owner to:
```

* Maintain strict access controls on your SCM provider (GitHub, GitLab)  
* Review and validate all code before it reaches production  
* Use protected branches and pull request reviews to enforce standards

> While roles can prevent a user from deploying an unvalidated branch directly from the dashboard they do not prevent a developer with access to your repository from introducing harmful logic into your codebase. We strongly recommend aligning your internal security practices with the deployment methods and access permissions you have configured on Scalingo.

Reviewer: > We strongly recommend aligning your internal security practices with the deployment methods and access permissions you have configured on Scalingo. <

Are we really encouraging our users to align their security practices with the Scalingo settings? Isn't it the opposite?
